enhancements/local-storage/automatic-detection.md

```diff
@@ -0,0 +1,492 @@
+	metav1.ListMeta `json:"metadata"`
+	Items           []LocalVolumeDiscovery `json:"items"`
+}
+
+type LocalVolumeDiscovery struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	Spec              LocalVolumeDiscoverySpec   `json:"spec"`
+	Status            LocalVolumeDiscoveryStatus `json:"status,omitempty"`
+}
+
+type LocalVolumeDiscoverySpec struct {
+	// Nodes on which the automatic detection policies must run.
+	// +optional
+	NodeSelector *corev1.NodeSelector `json:"nodeSelector,omitempty"`
+	// If specified tolerations is the list of toleration that is passed to the
+	// LocalVolumeDiscovery Daemon
+	// +optional
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+}
+
+type LocalVolumeDiscoveryPhase string
+
+const (
+	InProgressPhase LocalVolumeDiscoveryPhase = "InProgress"
+	FailedPhase     LocalVolumeDiscoveryPhase = "Failed"
+	CompletedPhase  LocalVolumeDiscoveryPhase = "Completed"
+)
+
+type LocalVolumeDiscoveryStatus struct {
+	Phase         LocalVolumeDiscoveryPhase `json:"phase,omitempty"`
+	CompletedJobs *int32                    `json:"completedjobs,omitempty"`
+	TotalJobs     *int32                    `json:"totaljobs,omitempty"`
+}
+
+type LocalVolumeNodeDiscoveryResultList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	Items           []LocalVolumeNodeDiscoveryResult `json:"items"`
+}
+
+type LocalVolumeNodeDiscoveryResult struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	Spec              LocalVolumeNodeDiscoveryResultSpec   `json:"spec"`
+	Status            LocalVolumeNodeDiscoveryResultStatus `json:"status,omitempty"`
+}
+
+type LocalVolumeNodeDiscoveryResultSpec struct {
+	NodeName string `json:"nodeName"`
+}
+
+type LocalVolumeDiscoveryResultPhase string
+
+const (
+	InProgressPhase LocalVolumeDiscoveryResultPhase = "Discovering"
+	FailedPhase     LocalVolumeDiscoveryResultPhase = "Failed"
+	CompletedPhase  LocalVolumeDiscoveryResultPhase = "Discovered"
+)
+
+type LocalVolumeNodeDiscoveryResultStatus struct {
+	Phase     LocalVolumeDiscoveryResultPhase `json:"phase,omitempty"`
+	TimeStamp string                          `json:"timestamp,omitempty"`
+	// Devices contains the list of devices on which LSO
+	// is capable of creating LocalPVs
+	// The devices in this list qualify these following conditions.
+	// - it should not have filesystem on it.
+	// - it should be a non-removable device.
+	// - it should not be a read-only device.
+	// - it should not be mounted anywhere
+	// - it should be a boot device
+	Devices []Device `json:"devices"`
+}
+
+type Device struct {
+	// Name of the detected device
+	Name string `json:"name"`
+	// Model of the detected device
+	Model string `json:"model"`
+	// Type of the detected device
+	Type string `json:"type"`
+	// Vendor of the detected device
+	Vendor string `json:"vendor"`
+	// UUID of the detected device
+	UUID string `json:"uuid"`
```

Comment:
UUID is actually not reliable for a raw disk. Let's remove it unless we can prove otherwise that it's useful.

Reply:
then how we will be creating the `LocalVolume` CR without `UUID`?